packages/vscode-extension/src/manifestListener.ts

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+"use strict";
+
+import * as vscode from "vscode";
+import {
+  isDeclarativeCopilotApp,
+  updateIsDeclarativeCopilotApp,
+  workspaceUri,
+} from "./globalVariables";
+import path from "path";
+import {
+  AppPackageFolderName,
+  ManifestTemplateFileName,
+  TeamsAppManifest,
+} from "@microsoft/teamsfx-api";
+import TreeViewManagerInstance from "./treeview/treeViewManager";
+import { ExtTelemetry } from "./telemetry/extTelemetry";
+import { TelemetryEvent, TelemetryProperty } from "./telemetry/extTelemetryEvents";
+import { isValidProjectV3 } from "@microsoft/teamsfx-core";
+
+function setAbortableTimeout(ms: number, signal: any) {
```

Comment:
You could take a look at [debounce](https://www.geeksforgeeks.org/lodash-_-debounce-method/) function.

Reply:
Thanks for letting me know the alternative :)